OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -172,8 +172,31 @@ public interface IVoiced
 		bool HasVoice(Actor self, string voice);
 	}
 
-	[RequireExplicitImplementation]
-	public interface IStoreResources { int Capacity { get; } }
+	public interface IStoresResourcesInfo : ITraitInfoInterface
+	{
+		string[] ResourceTypes { get; }
+	}
+
+	public interface IStoresResources
```

Comment:
This should also be explicitly implemented.

Reply:
Why? It would just force nasty code in other implementations

Reply:
All our interfaces should be explicit. Why would this force nasty code?

Reply:
because if the interface is explicit, you can't reference members within the class